pitop/camera/camera.py

```diff
@@ -14,20 +14,30 @@ class Camera:
     Provides a variety of high-level functionality for using the PMA USB Camera, including capturing
     images and video, and processing image data from the camera
 
-    :type camera_device_id: int
-    :param camera_device_id:
-        The ID for the port to which this component is connected. Defaults to 0.
+    :type index: int
+    :param index:
+        ID of the video capturing device to open.
+        To open default camera using default backend just pass 0.
     """
 
-    def __init__(self, camera_device_id=0, camera_type=CameraTypes.USB_CAMERA, path_to_images=""):
+    def __init__(self,
+                 index=0,
+                 resolution=None,
+                 camera_type=CameraTypes.USB_CAMERA,
+                 path_to_images="",
+                 format='PIL'
+                 ):
 
         if camera_type == CameraTypes.USB_CAMERA:
             from .core import UsbCamera
-            self.__camera = UsbCamera(camera_device_id)
+            self.__camera = UsbCamera(index, resolution)
         elif camera_type == CameraTypes.FILE_SYSTEM_CAMERA:
             from .core import FileSystemCamera
             self.__camera = FileSystemCamera(path_to_images)
 
+        # Frame callback
+        self.on_frame = None
+        self._on_frame_image_format = format
```

Comment:
Basic check - is it at least PIL or OpenCV? if not, things are gonna be bad, mkay